storage-proofs/porep/src/stacked/vanilla/proof_scheme.rs

```diff
@@ -154,6 +154,10 @@ impl<'a, 'c, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> ProofScheme<'
     ) -> bool {
         let partition_challenges = public_params.layer_challenges.challenges_count_all();
 
+        assert_eq!(
+            partition_challenges.checked_mul(partitions),
+            Some(partition_challenges * partitions)
+        );
```

Comment:
Technically, it would be okay if this overflows because the result would be conservative and, if anything, lead to requirements being judged unsatisfied when they 'should' be satisfied. Of course this cannot happen for correct protocol values anyway, and I agree a check is helpful for development.

That said, if this 'error' were hit in reality, it would probably mean we need to revise the code. Technically, values which would overflow could be 'correct'. None of this is a practical issue. I'm just thinking through the significance of the assertion.

I don't recommend or request it, but we could 'technically' return `true` if the assertion fails (even though this would undoubtedly be papering over a usage error).

Reply:
For now, I'll leave in the assert as-is, but realize they should not trigger in normal circumstances and the downside is that we're calculating values more than needed.  If we return true if the check fails, there's probably no reason to do it since I think even logging it would likely be lost/ignored.